include/rocksdb/cache.h

```diff
@@ -534,40 +545,21 @@ class Cache {
   //
   // When the inserted entry is no longer needed, the key and
   // value will be passed to "deleter".
-  virtual Status Insert(const Slice& key, void* value,
+  virtual Status Insert(const Slice& key, ValueType* value,
```

Comment:
If the doesn't want the item to be demoted to the secondary cache i.e if `DBOptions::lowest_used_cache_tier` is set to the volatile tier, I guess they should pass a `CacheItemHelper` with nullptr callbacks? Would be good to clarify in the comments.